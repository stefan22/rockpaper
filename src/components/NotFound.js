import React from 'react';
import {Link} from 'react-router-dom';
import '../scss/not-found.scss';

const NotFound = () => (

  <div className='not-found'>
    <h2 className='not-found-title'>Page Not Found</h2>
    <Link className='not-found-link' to='/'>Back to Homepage</Link>
  </div>

);


export default NotFound;
