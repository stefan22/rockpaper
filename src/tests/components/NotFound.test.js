import React from 'react';
import {shallow} from 'enzyme';
import NotFound from '../../components/NotFound';
import toJSON from 'enzyme-to-json';

describe('NotFound page component', () => {
  it('should render NotFound page correctly', () => {
    const wrapper = shallow(<NotFound />);
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  it('should have class of "not-found"', () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper.find('.not-found').length).toBe(1);
  });
  it('should have text content of "Page Not Found"', () => {
    const wrapper = shallow(<NotFound />);
    expect(wrapper.text()).toEqual('Page Not Found<Link />');
  });


});
