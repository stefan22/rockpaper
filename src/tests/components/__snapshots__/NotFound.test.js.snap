// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound page component should render NotFound page correctly 1`] = `
<div
  className="not-found"
>
  <h2
    className="not-found-title"
  >
    Page Not Found
  </h2>
  <Link
    className="not-found-link"
    replace={false}
    to="/"
  >
    Back to Homepage
  </Link>
</div>
`;
